addons/stock/wizard/stock_picking_return.py

```diff
@@ -95,9 +95,10 @@ def _create_returns(self):
             'move_lines': [],
             'picking_type_id': picking_type_id,
             'state': 'draft',
-            'origin': self.picking_id.name,
+            'origin': _("Return of %s") % self.picking_id.name,
```

Comment:
@ppr-odoo wouldn't it be better to do `'origin': _("Return of ")` instead of `'origin': _("Return of %s") % self.picking_id.name,`?
Otherwise you would need to write (or have) a translation for every specific picking record name. If `Return of` is translated the name will be filled in by the user itself so should be fine by itself. (cc @xmo-odoo)

Reply:
@Yenthe666 no the current version is correct, different grammar may need to decorate differently e.g. they may put the name first or in the middle or whatever. Using a format string is the correct way to do that.

Reply:
@xmo-odoo thank you for the feedback. Actually, I've seen a lot of concatenations around Odoo that is why I was under the assumption that would be better. Sorry for the noise.

Reply:
@Yenthe666 no worries, as I said concatenations of a translated string and a programmatic value is something we do way too much and which is often problematic, even more so sequences e.g. `_("a bit") + a_value + _("an other bit") + an_other_vaue` becomes very difficult to translate as not only the order can be completely unsuitable there isn't enough context, and the translation could differ between the individual bits and the bits *as part of a broader sentence*.

[Here's an example I saw this morning](https://thegeekygaeilgeoir.wordpress.com/2017/09/06/even-racists-got-the-blues/) of somebody doing word-by-word translation of english to irish yielding absolute nonsense, the result figuring incorrect grammar and incorrect individual words *for the context* as well as literally translating a non-cross-cultural idiom.

Reply:
@xmo-odoo thanks a lot for the further clarification. The article clearly pointed out why this is really important - an eye opener. 😉